data-prepper-api/src/main/java/org/opensearch/dataprepper/model/event/JacksonEvent.java

```diff
@@ -373,19 +373,9 @@ private String trimKey(final String key) {
     }
 
     private boolean isValidKey(final String key) {
-        char previous = ' ';
-        char next = ' ';
         for (int i = 0; i < key.length(); i++) {
             char c = key.charAt(i);
 
-            if (i < key.length() - 1) {
-                next = key.charAt(i + 1);
-            }
-
-            if ((i == 0 || i == key.length() - 1 || previous == '/' || next == '/') && (c == '_' || c == '.' || c == '-')) {
-                return false;
```

Comment:
What was the motivation for not allowing these characters initially? I'm wondering what the consequences of this change would be

Reply:
I am not sure what the reasoning was

Reply:
This is the PR where I added this custom code (https://github.com/opensearch-project/data-prepper/pull/725/files). But the regex pattern that was used to validate was added by @cmanning09